sky/skylet/providers/runpod/rp_helper.py

```diff
@@ -0,0 +1,116 @@
+'''
+RunPod library wrapper, formats the input/output of the RunPod library for SkyPilot.
+'''
+import os
+import json
+from typing import Dict
+
+import runpod
+
+
+GPU_NAME_MAP = {
+    "A100-80GB": "NVIDIA A100 80GB PCIe",
+    "A100-40GB": "NVIDIA A100-PCIE-40GB",
+    "A100-80GB-SXM4": "NVIDIA A100-SXM4-80GB",
+    "A30": "NVIDIA A30",
+    "A40": "NVIDIA A40",
+    "RTX3070": "NVIDIA GeForce RTX 3070",
+    "RTX3080": "NVIDIA GeForce RTX 3080",
+    "RTX3080Ti": "NVIDIA GeForce RTX 3080 Ti",
+    "RTX3090": "NVIDIA GeForce RTX 3090",
+    "RTX3090Ti": "NVIDIA GeForce RTX 3090 Ti",
+    "RTX4070Ti": "NVIDIA GeForce 4070 Ti",
+    "RTX4080": "NVIDIA GeForce RTX 4080",
+    "RTX4090": "NVIDIA GeForce RTX 4090",
+    "H100-80GB-HBM3": "NVIDIA H100 80GB HBM3",
+    "H100-PCIe": "NVIDIA H100 PCIe",
+    "L40": "NVIDIA L40",
+    "RTX6000-Ada": "NVIDIA RTX 6000 Ada Generation",
+    "RTXA4000": "NVIDIA RTX A4000",
+    "RTXA4500": "NVIDIA RTX A4500",
+    "RTXA5000": "NVIDIA RTX A5000",
+    "RTXA6000": "NVIDIA RTX A6000",
+    "RTX5000": "Quadro RTX 5000",
+    "V100-16GB-FHHL": "Tesla V100-FHHL-16GB",
+    "V100-16GB-SXM2": "V100-SXM2-16GB"
+}
+
+
+TAG_FILE = os.path.expanduser('~/.runpod/skypilot_tags.json')
+
+if not os.path.exists(TAG_FILE):
+    with open(TAG_FILE, 'w') as tag_file:
+        json.dump({}, tag_file)
+
+
+def list_instances():
+    '''
+    Lists instances associated with API key.
+    '''
+    instances = runpod.get_pods()
+
+    instance_list = {}
+    for instance in instances:
+        instance_list[instance['id']] = {}
+
+        instance_list[instance['id']]['status'] = instance['desiredStatus']
+        instance_list[instance['id']]['name'] = instance['name']
+
+        if instance['desiredStatus'] == 'running':
+            for port in instance['runtime']['ports']:
+                if port['privatePort'] == 22:
+                    instance_list[instance['id']]['ip'] = port['ip']
```

Comment:
It seems the following instance config contains both private and public IPs. Should we have both information in the instance_list, as it is required by the caller?
```
{'id': 'ka1kxqqpvja5hz', 'containerDiskInGb': 5, 'costPerHr': 0.34, 'desiredStatus': 'RUNNING', 'dockerArgs': None, 'dockerId': None, 'env': [], 'gpuCount': 1, 'imageName': 'runpod/pytorch:2.0.1-py3.10-cuda11.8.0-devel', 'lastStatusChange': 'Rented by User: Tue Sep 05 2023 07:16:20 GMT+0000 (Coordinated Universal Time)', 'machineId': 'd20416f3469v', 'memoryInGb': 23, 'name': 'sky-aace-gcpuser', 'podType': 'RESERVED', 'port': None, 'ports': '22/tcp', 'uptimeSeconds': 0, 'vcpuCount': 6, 'volumeInGb': 0, 'volumeMountPath': '/workspace', 'runtime': {'ports': [{'ip': '100.65.10.182', 'isIpPublic': False, 'privatePort': 19123, 'publicPort': 60011, 'type': 'http'}, {'ip': '149.36.0.160', 'isIpPublic': True, 'privatePort': 22, 'publicPort': 48246, 'type': 'tcp'}]}, 'machine': {'gpuDisplayName': 'RTX A4000'}}
```